+sw_tests/+unit_tests/unittest_spinw_spinwave.m

```diff
@@ -450,31 +493,40 @@ function test_hermit(testCase)
             spec = swobj_h.spinwave(hkl, 'hermit', false);
             testCase.assertGreaterThan(sum(abs(imag(spec.omega(:)))), 0);
         end
-        function test_sw_qh5_tol(testCase)
+        function test_sw_qh5_tol(testCase, mex)
+            swpref.setpref('usemex', mex);
             tol = 5e-4;
             qpts = testCase.qh5;
             swobj_tol = copy(testCase.swobj);
             % Generate magstr that deviates slightly from commensurate
             swobj_tol.genmagstr('mode', 'helical', 'k', [tol 0 0], ...
                                 'n', [0 0 1], 'S', [0; 1; 0]);
-            % Check that without tol it is incommensurate - causes error
-            testCase.verifyError(...
-                @() swobj_tol.spinwave(qpts), ...
-                'spinw:spinwave:NonPosDefHamiltonian');
+            if ~mex
+                % Check that without tol it is incommensurate - causes error
+                testCase.verifyError(...
+                    @() swobj_tol.spinwave(qpts), ...
+                    'spinw:spinwave:NonPosDefHamiltonian');
```

Comment:
Weird that this non-positive-definite error doesn't come up in the mex file, but it doesn't - oh well...